### PR TITLE
Write Unit Testing

### PR DESCRIPTION
**Pull request by Alfred-KInyua**
Set up React Testing Library.
Write unit tests for the files operate.js and calculate.js using Jest.
Create unit tests for all React components:
Write unit tests with React Testing Library.
Write unit tests with snapshots with Jest.

### CODE REVIEW

File: src/components/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// import { render,screen } from "@testing-library/react";
```
Comment (danmainah) [inline src/components/logic/calculate.test.js]: @Alfred-KInyua and @ChrispaixK great work on your tests, kindly ensure you don't have commented coed in your codebase its not good practice.

File: src/components/test/Calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// import { Renderer } from 'react-test-renderer';
```
Comment (danmainah) [inline src/components/test/Calculator.test.js]: @Alfred-KInyua and @ChrispaixK great work on your tests, kindly ensure you don't have commented coed in your codebase its not good practice.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Alfred-KInyua and @ChrispaixK , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/test/Calculator.test.js (hunk 2 above)
Comment (Alfred-KInyua) [reply src/components/test/Calculator.test.js]: @danmainah Thanks for your review.

Comment (ChrispaixK) [reply src/components/test/Calculator.test.js]: Thank you @danmainah

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**